internal/dbutil/sqliteutil/migrator.go

```diff
@@ -12,63 +12,66 @@ import (
 	"github.com/zeebo/errs"
 )
 
-// MigrateTablesToDatabase copies the specified tables from srcDB into destDB.
-// All tables in destDB will be dropped other than those specified in
-// tablesToKeep.
-func MigrateTablesToDatabase(ctx context.Context, srcDB, destDB *sql.DB, tablesToKeep ...string) error {
-	// Retrieve the raw Sqlite3 driver connections for the src and dest so that
-	// we can execute the backup API for a corruption safe clone.
-	srcConn, err := srcDB.Conn(ctx)
-	if err != nil {
-		return errs.Wrap(err)
-	}
-
-	destConn, err := destDB.Conn(ctx)
-	if err != nil {
-		return errs.Wrap(err)
-	}
+var (
+	// ErrMigrateTables is error class for MigrateTables
+	ErrMigrateTables = errs.Class("migrate tables:")
 
-	// The references to the driver connections are only guaranteed to be valid
-	// for the life of the callback so we must do the work within both callbacks.
-	err = srcConn.Raw(func(srcDriverConn interface{}) error {
-		srcSqliteConn, ok := srcDriverConn.(*sqlite3.SQLiteConn)
-		if !ok {
-			return errs.New("unable to get database driver")
-		}
-
-		err = destConn.Raw(func(destDriverConn interface{}) error {
-			destSqliteConn, ok := destDriverConn.(*sqlite3.SQLiteConn)
-			if !ok {
-				return errs.New("unable to get database driver")
-			}
+	// ErrKeepTables is error class for MigrateTables
+	ErrKeepTables = errs.Class("keep tables:")
+)
 
-			err = backup(ctx, srcSqliteConn, destSqliteConn)
-			if err != nil {
-				return errs.New("unable to backup database")
-			}
-			return nil
-		})
+// MigrateTables copies the specified tables from srcDB into destDB.
+// All tables in destDB will be dropped other than those specified in
+// tablesToKeep.
+func MigrateTables(ctx context.Context, srcDB, destDB *sql.DB, tablesToKeep ...string) error {
+	err := func() error {
```

Comment:
I'm not liking this closure, but it's the shortest way to write this without introducing an extra func.